### PR TITLE
[WebProfilerBundle] Add link to show profile of latest request

### PR DESCRIPTION
**Pull request by xelaris**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

This PR adds a "View latest" link next to "View last 10" to enable quick access to the profile of the latest request. This is useful especially when using the profiler along with a REST client. After sending a new request it is now possible to view the latest profile with just one click.

### CODE REVIEW

**Comment by aitboudad** [comment]
:+1: you need also change https://github.com/symfony/symfony/blob/2.7/src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/info.html.twig#L30 when token == latest

**Comment by xelaris** [comment]
Thanks for the review. The PR is updated to show a more precise message, if there are no profiles.

File: src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php [hunk 1]
```diff
@@ -91,6 +91,11 @@ public function panelAction(Request $request, $token)
         $panel = $request->query->get('panel', 'request');
         $page = $request->query->get('page', 'home');
 
+        if ('latest' === $token) {
+            $latest = current($this->profiler->find(null, null, 1, null, null, null));
+            $token = $latest ? $latest['token'] : $token;
+        }
```
Comment (fabpot) [inline src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php]: I would simplify the code to:

```php
        if ('latest' === $token && $latest = current($this->profiler->find(null, null, 1, null, null, null))) {
            $token = $latest['token'];
        }
```

**Comment by xelaris** [comment]
@fabpot I changed the code as suggested.

**Comment by xelaris** [comment]
I think this can be merged in 2.8 as it is, can't it?

**Comment by fabpot** [comment]
Thank you @xelaris.

**Pull request closed without merging**